units/host_allocator.go

```diff
@@ -198,27 +198,34 @@ func (j *hostAllocatorJob) Run(ctx context.Context) {
 		"duration_secs": time.Since(hostSpawningBegins).Seconds(),
 	})
 
-	var makespan time.Duration
-	if len(hostsSpawned) != 0 {
-		makespan = distroQueueInfo.ExpectedDuration / time.Duration(len(hostsSpawned))
-	} else if distroQueueInfo.Length > 0 {
-		makespan = distroQueueInfo.ExpectedDuration
-	}
+	// ignoring all the tasks that will take longer than the threshold to run,
+	// and the hosts allocated for them,
+	// how long will it take the current fleet of hosts, plus the ones we spawned, to chew through
+	// the scheduled tasks in the queue?
+	scheduledDuration := distroQueueInfo.ExpectedDuration - distroQueueInfo.DurationOverThreshold
+
+	timeToEmpty := scheduledDuration / time.Duration((nHostsFree + len(hostsSpawned) - distroQueueInfo.CountDurationOverThreshold))
+	timeToEmptyNoSpawns := scheduledDuration / time.Duration((nHostsFree - distroQueueInfo.CountDurationOverThreshold))
+
+	hostQueueRatio := time.Duration(timeToEmpty) / distroQueueInfo.MaxDurationThreshold
+	noSpawnsRatio := time.Duration(timeToEmptyNoSpawns) / distroQueueInfo.MaxDurationThreshold
 
 	grip.Info(message.Fields{
-		"message":                 "distro-scheduler-report",
-		"job_type":                hostAllocatorJobName,
-		"distro":                  distro.Id,
-		"provider":                distro.Provider,
-		"max_hosts":               distro.HostAllocatorSettings.MaximumHosts,
-		"num_new_hosts":           len(hostsSpawned),
-		"pool_info":               existingHosts.Stats(),
-		"queue":                   eventInfo,
-		"total_runtime":           distroQueueInfo.ExpectedDuration.String(),
-		"runtime_secs":            distroQueueInfo.ExpectedDuration.Seconds(),
-		"predicted_makespan":      makespan.String(),
-		"predicted_makespan_secs": makespan.Seconds(),
-		"instance":                j.ID(),
-		"runner":                  scheduler.RunnerName,
+		"message":                    "distro-scheduler-report",
+		"job_type":                   hostAllocatorJobName,
+		"distro":                     distro.Id,
+		"provider":                   distro.Provider,
+		"max_hosts":                  distro.HostAllocatorSettings.MaximumHosts,
+		"num_new_hosts":              len(hostsSpawned),
+		"pool_info":                  existingHosts.Stats(),
+		"queue":                      eventInfo,
+		"total_runtime":              distroQueueInfo.ExpectedDuration.String(),
+		"runtime_secs":               distroQueueInfo.ExpectedDuration.Seconds(),
+		"time_to_empty":              timeToEmpty.String(),
+		"time_to_empty_no_spawns":    timeToEmptyNoSpawns.String(),
+		"host_queue_ratio":           hostQueueRatio.String(),
```

Comment:
These 2 are ratios, but calling the string function on a duration will display "ns" after them. I think you want to do Nanoseconds() instead

Reply:
Oh wait actually since these are ints, you'll want to convert to floating point numbers before doing the division

Reply:
Thanks for the catch! I modified it to convert to ns first, then divide, so we can get the float behavior I wanted.